frontend/src/scenes/feature-flags/featureFlagLogic.ts

```diff
@@ -688,6 +679,58 @@ export const featureFlagLogic = kea<featureFlagLogicType>([
             },
         ],
     }),
+    sharedListeners(({ values }) => ({
+        checkDependentFlagsAndConfirm: async (payload: {
+            originalFlag: FeatureFlagType | null
+            updatedFlag: Partial<FeatureFlagType>
+            onConfirm: () => void
+        }) => {
+            const { originalFlag, updatedFlag, onConfirm } = payload
+
+            // Check if flag is being disabled and has active dependents
+            const isBeingDisabled = updatedFlag.id && originalFlag?.active === true && updatedFlag.active === false
+
+            let dependentFlagsWarning: string | undefined
+            if (isBeingDisabled) {
+                try {
+                    const response = await api.create(
+                        `api/projects/${values.currentProjectId}/feature_flags/${updatedFlag.id}/has_active_dependents/`,
+                        {}
+                    )
+                    if (response.has_active_dependents) {
+                        dependentFlagsWarning = response.warning
+                    }
+                } catch {
+                    lemonToast.error('Failed to check for dependent flags. Please try again.')
+                    return
+                }
+            }
+
+            const customMessages: string[] = []
+            if (dependentFlagsWarning) {
+                customMessages.push(dependentFlagsWarning)
+            }
+            if (values.currentTeam?.feature_flag_confirmation_enabled) {
+                const teamCustomMessage = values.currentTeam?.feature_flag_confirmation_message
+                if (teamCustomMessage) {
+                    customMessages.push(teamCustomMessage)
+                }
+            }
+
+            const confirmationShown = checkFeatureFlagConfirmation(
+                originalFlag,
+                updatedFlag as FeatureFlagType,
+                customMessages.length > 0,
```

Comment:
This seems off to me. Custom messages are optional. If they enable confirmation messages, we should show this, right?

Maybe we could define a variable next to where we define `messagesToDisplay` and use it here:

```ts
const needsConfirmation =
      values.currentTeam?.feature_flag_confirmation_enabled ||
      customMessages.length > 0
```